lib/eventsource.js

```diff
@@ -272,8 +356,17 @@ function EventSource (url, eventSourceInitDict) {
     })
 
     req.on('error', function (err) {
-      self.connectionInProgress = false
-      onConnectionClosed(err.message)
+      onConnectionFailed(err.message)
+    })
+    // FF Server sends keepalive messages every 20 seconds, so timeout on 30 seconds to account for latency.
+    var timeout = 30000
+    req.setTimeout(timeout)
+
+    req.on('timeout', function () {
+      onConnectionFailed({
+        message: 'Read timeout, received no data in ' + timeout +
+          'ms, assuming connection is dead'
```

Comment:
Not a problem, just an FYI that template strings are great for this kinda thing
```suggestion
        message: `Read timeout, received no data in ${timeout}ms, assuming connection is dead`
```